docs/spark_expressions_support.md

```diff
@@ -0,0 +1,475 @@
+ - [x] date_part
+ - [ ] date_sub
+ - [ ] date_trunc
+ - [ ] dateadd
+ - [ ] datediff
+ - [x] datepart
+ - [ ] day
+ - [ ] dayofmonth
+ - [ ] dayofweek
+ - [ ] dayofyear
+ - [x] extract
+ - [ ] from_unixtime
+ - [ ] from_utc_timestamp
+ - [ ] hour
+ - [ ] last_day
+ - [ ] localtimestamp
+ - [ ] make_date
+ - [ ] make_dt_interval
+ - [ ] make_interval
+ - [ ] make_timestamp
+ - [ ] make_timestamp_ltz
+ - [ ] make_timestamp_ntz
+ - [ ] make_ym_interval
+ - [ ] minute
+ - [ ] month
+ - [ ] months_between
+ - [ ] next_day
+ - [ ] now
+ - [ ] quarter
+ - [ ] second
+ - [ ] timestamp_micros
+ - [ ] timestamp_millis
+ - [ ] timestamp_seconds
+ - [ ] to_date
+ - [ ] to_timestamp
+ - [ ] to_timestamp_ltz
+ - [ ] to_timestamp_ntz
+ - [ ] to_unix_timestamp
+ - [ ] to_utc_timestamp
+ - [ ] trunc
+ - [ ] try_to_timestamp
+ - [ ] unix_date
+ - [ ] unix_micros
+ - [ ] unix_millis
+ - [ ] unix_seconds
+ - [ ] unix_timestamp
+ - [ ] weekday
+ - [ ] weekofyear
+ - [ ] year
+
+### generator_funcs
+ - [ ] explode
+ - [ ] explode_outer
+ - [ ] inline
+ - [ ] inline_outer
+ - [ ] posexplode
+ - [ ] posexplode_outer
+ - [ ] stack
+
+### hash_funcs
+ - [ ] crc32
+ - [ ] hash
+ - [x] md5
+ - [ ] sha
+ - [ ] sha1
+ - [ ] sha2
+ - [ ] xxhash64
+
+### json_funcs
+ - [ ] from_json
+ - [ ] get_json_object
+ - [ ] json_array_length
+ - [ ] json_object_keys
+ - [ ] json_tuple
+ - [ ] schema_of_json
+ - [ ] to_json
+
+### lambda_funcs
+ - [ ] aggregate
+ - [ ] array_sort
+ - [ ] exists
+ - [ ] filter
+ - [ ] forall
+ - [ ] map_filter
+ - [ ] map_zip_with
+ - [ ] reduce
+ - [ ] transform
+ - [ ] transform_keys
+ - [ ] transform_values
+ - [ ] zip_with
+
+### map_funcs
+ - [ ] element_at
+ - [ ] map
+ - [ ] map_concat
+ - [ ] map_contains_key
+ - [ ] map_entries
+ - [ ] map_from_arrays
+ - [ ] map_from_entries
+ - [ ] map_keys
+ - [ ] map_values
+ - [ ] str_to_map
+ - [ ] try_element_at
+
+### math_funcs
+ - [x] %
+ - [x] *
+ - [x] +
+ - [x] -
+ - [x] /
+ - [x] abs
+ - [x] acos
+ - [ ] acosh
+ - [x] asin
+ - [ ] asinh
+ - [x] atan
+ - [x] atan2
+ - [ ] atanh
+ - [ ] bin
+ - [ ] bround
+ - [ ] cbrt
+ - [x] ceil
+ - [x] ceiling
+ - [ ] conv
+ - [x] cos
+ - [ ] cosh
+ - [ ] cot
+ - [ ] csc
+ - [ ] degrees
+ - [ ] div
+ - [ ] e
+ - [x] exp
+ - [ ] expm1
+ - [ ] factorial
+ - [x] floor
+ - [ ] greatest
+ - [ ] hex
+ - [ ] hypot
+ - [ ] least
+ - [x] ln
+ - [ ] log
+ - [x] log10
+ - [ ] log1p
+ - [x] log2
+ - [x] mod
+ - [x] negative
+ - [ ] pi
+ - [ ] pmod
+ - [x] positive
+ - [x] pow
+ - [x] power
+ - [ ] radians
+ - [ ] rand
+ - [ ] randn
+ - [ ] random
+ - [ ] rint
+ - [x] round
+ - [ ] sec
+ - [x] shiftleft
+ - [x] sign
+ - [x] signum
+ - [x] sin
+ - [ ] sinh
+ - [x] sqrt
+ - [x] tan
+ - [ ] tanh
+ - [x] try_add
+ - [x] try_divide
+ - [x] try_multiply
+ - [x] try_subtract
+ - [x] unhex
+ - [ ] width_bucket
+
+### misc_funcs
+ - [ ] aes_decrypt
+ - [ ] aes_encrypt
+ - [ ] assert_true
+ - [x] current_catalog
+ - [x] current_database
+ - [x] current_schema
+ - [x] current_user
```

Comment:
We don't have a native implementation for these methods. I am guessing that the Spark planner replaces these with literal values and we then see a CometProject for those literal values

Reply:
That's correct. These expressions are replaced with literals after analysis stage.

Reply:
Yes, those are purely spark execution, are we planning to natively support them in Comet?

Reply:
We cannot support these in Comet because these functions will never appear in the physical plan. Spark replaces them with literal values before this.

Reply:
sounds good, I'll remove them from the list